### ISSUE TITLE
Tests don't fail on type failures

### ISSUE THREAD

**Issue by saulshanabrook** [jupyterlab/jupyterlab#4710]
If you have broken types in the test files, the tests will still run and could pass. Instead, they shouldn't run at all. We attempted to address this in #4698 but couldn't find a way to make them fail.

**Comment by jasongrout** [comment]
Basically, this happens because we check types in parallel after we transpile the code. We didn't figure out a way for the type-checker to error and fail the test.

**Comment by blink1073** [comment]
cf https://github.com/webpack-contrib/karma-webpack/issues/66, which is why I we need to convert to Jest.   Closing in favor of #5173
